lib/sax.js

```diff
@@ -1562,4 +1145,460 @@
       }
     }())
   }
+
```

Comment:
This is the beginning of the functions that were extracted from the original `switch` statement. For the most part they are just copied/pasted from their original location w/ little to no alteration (other then changing `continue` to `return` in a few places)